### ISSUE TITLE
How does Shorebird do the magic

### ISSUE THREAD

**Issue by macwilko** [shorebirdtech/shorebird#871]
I’m really curious to know how achieved OTA updates+ Flutter. 

What do you do to make this magic possible. 

Are there any trade offs or disadvantages to the performance of a Flutter app using Shorebird? 

Thank-you kindly, greatly anticipating Windows and macOS support.

**Comment by eseidel** [comment]
Yeah, I need to write up docs on this.  I think I have pieces in places.

On Android we just tell the Dart VM to load different Dart code at runtime.
On iOS we've modified the Dart VM to use an interpreter + a fancy linker.
Both are done that way to comply with the relevant stores.  Some of which I spoke about in my FlutterCon talk (video will hopefully be posted soon).
https://docs.google.com/presentation/d/1MlJut_BIdqY6VjR-ota5pZ2lGw7ROWX-wRwTERKiRnQ/edit#slide=id.p

There is zero performance change on Android.

There is currently a performance change on iOS because we've only enabled our interpreter (the compliance part of our solution) the "go-fast" part of our solution "linker" will change it from running *all* code in our interpreter to running *only changed* code in our interpreter and otherwise using the pre-signed code in your binary for most things.  We have to finish teaching the Dart VM about this "mixed mode" first.

I intend to write up some longer articles once we get iOS out.  https://github.com/shorebirdtech/docs/issues/99

https://github.com/shorebirdtech/shorebird/issues/397 tracks desktop support.  We won't get to that until after iOS is out at least.

Thanks for reaching out!  Hope that helps.  We're available on discord every day: https://discord.gg/shorebird
